### DISCUSSION TITLE
86Box build steps using MSYS2 on Windows v0.5 by saipan7

### DISCUSSION THREAD

**Discussion by ghost** [86Box/86Box#3700] [Show and tell]
> [!WARNING]
> This guide is outdated, please follow the official build guide instead: https://86box.readthedocs.io/en/latest/dev/buildguide.html

<details>

<summary>Original guide for archival purposes</summary>
86Box build steps using MSYS2 on Windows v0.5 by saipan7

source:
https://86box.readthedocs.io/en/latest/dev/buildguide.html
https://github.com/86Box/86Box/blob/master/.github/workflows/cmake.yml

1. download and install:

	https://github.com/msys2/msys2-installer/releases/download/2025-02-21/msys2-x86_64-20250221.exe

- on the opened MSYS2 window, run below command to update the MSYS2: 

	pacman -Syu

- choose Y, when requested to close the MSYS2 window to apply the pending updates.
- open new opened MSYS2 window, run below command to check for update again for the MSYS2: 

	pacman -Syu

- close the MSYS2 window. 

2. Open new MSYS2 MINGW64, run below command:

	pacman -Syu
	pacman -S pactoys
	pacboy -S ninja
	pacboy -S cmake
	pacboy -S gcc
	pacboy -S pkgconf
	pacboy -S freetype
	pacboy -S SDL2
	pacboy -S zlib
	pacboy -S libpng
	pacboy -S openal
	pacboy -S rtmidi
	pacboy -S libslirp
	pacboy -S fluidsynth
	pacboy -S libserialport
	pacboy -S libvncserver
	pacboy -S qt5-static
        pacboy -S vulkan-headers

3. Open new MSYS2 window, run below command to download and install the git client:

	pacboy -S git

OPTIONAL: close the MSYS2 window, zip backup your C:\msys64 folder and name it as "msys64 (86box base build image).zip", 
just incase you want to start all over again, you can then delete your C:\msys64 folder and then restore this zip image to c:\ 
and the use it as starting point to build 86box.

4. Open new MSYS2 MINGW64 window, run below command to download 86Box source code:

	git clone https://github.com/86Box/86Box.git

5. On MSYS2 MINGW64, run below command to compile 86Box source code to 86Box binary files

	cmake -G Ninja -B ~/86Box/build -S ~/86Box --preset regular

6. On MSYS2 MINGW64, run below command to convert 86Box binary files to 86Box.exe

	cmake --build ~/86Box/build

7. On MSYS2 MINGW64, run below command to strip the debug info from the 86Box.exe and then copy it to the ~/86Box/build/artifacts folder

	cmake --install ~/86Box/build --prefix ~/86Box/build/artifacts --strip
	
	
8. 86Box.exe executable ready to be use inside the ~/86Box/build/artifacts folder.

![image](https://github.com/86Box/86Box/assets/127716019/6e31a5ff-3d79-41f7-912f-900d3ac9e666)
</details>

**Comment by ghost** [comment]
documentation & log files:

[86Box build steps using MSYS2 MINGW32 in Windows v0.5.txt](https://github.com/86Box/86Box/files/12684443/86Box.build.steps.using.MSYS2.MINGW32.in.Windows.v0.5.txt)

[86Box build steps using MSYS2 MINGW32 in Windows v0.5 log.txt](https://github.com/86Box/86Box/files/12684445/86Box.build.steps.using.MSYS2.MINGW32.in.Windows.v0.5.log.txt)

**Comment by ghost** [comment]
How to build 86Box specific source code branch


To download and build specific 86Box branch source code, run below code on MSYS2 MINGW32.
As an example, the below code will download and then build the 86Box from the "version/4.1" source code branch.
If you want to build the 86Box from other source code branch, just replace the "version/4.1" branch name to 
any 86Box branch name you like which available on the 86Box source code tree.

here are the codes:

git clone https://github.com/86Box/86Box.git --branch version/4.1 --single-branch 86box

cmake -G Ninja -B ~/86Box/build -S ~/86Box --preset regular
cmake --build ~/86Box/build
cmake --install ~/86Box/build --prefix ~/86Box/build/artifacts --strip


![image](https://github.com/86Box/86Box/assets/127716019/f06236c8-7a69-4ecc-ab94-4f82bcb133d4)

86Box.exe executable ready to be use inside the ~/86Box/build/artifacts folder and 
with the correct roms from the 86Box branch you choose, 
now you can access new feature of the newly generated 86Box executable.
 
![image](https://github.com/86Box/86Box/assets/127716019/97b737a0-da5f-464f-8af6-95249d962b91)

**Comment by ghost** [comment]
[Updated MSYS2 MINGW64 build shell script for windows  (14 Dec 2024).zip](https://github.com/user-attachments/files/18135607/Updated.MSYS2.MINGW64.build.shell.script.for.windows.14.Dec.2024.zip)


Regular build example run and output location:
![86box regular build](https://github.com/user-attachments/assets/70e35bd6-1e64-43a0-918a-ad60475fbd28)
![86box regular build](https://github.com/user-attachments/assets/00f307fa-382a-4438-9d6c-9ce820427782)

Debug build example run and output location:
![86box debug build](https://github.com/user-attachments/assets/38fd7a05-f971-4eed-b00c-5adbcf5cbe37)
![86box debug build](https://github.com/user-attachments/assets/a653ef8f-4f29-4919-91c4-d65bfc6ebf63)

**Comment by Iamblichos** [comment]
Hi, I got a target not found on freetype, and many other packages after running the pacman command.

**Reply by jriwanek** [reply]
look at the official build guide at https://86box.readthedocs.io/en/latest/dev/buildguide.html
